msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -7728,12 +7729,17 @@ private DiscussionTopicBean duplicateTopic(Long originalTopicId, DiscussionForum
 			log.debug("About to get a member of membershipIter");
 			DBMembershipItem oldItem = (DBMembershipItem)membershipIter.next();
 				log.debug("About to getMembershipItemCopy()");
-				DBMembershipItem newItem = getMembershipItemCopy(oldItem);
-				if (newItem != null) {
-					newItem = permissionLevelManager.saveDBMembershipItem(newItem);
-					newTopic.addMembershipItem(newItem);
+				DBMembershipItem itemCopy = getMembershipItemCopy(oldItem);
+				if (itemCopy != null) {
+					DBMembershipItem newItem = permissionLevelManager.saveDBMembershipItem(itemCopy);
+					membershipItemSet.add(newItem);
 				}
 		}
+
+		// This will bond the new permission items to the topic with surrogate key
+		newTopic.setMembershipItemSet(membershipItemSet);
+		DiscussionTopic finalNewTopic = newTopic;
```

Comment:
assuming you need the final because of the lambda why not just make newTopic final which eliminates the extra var?